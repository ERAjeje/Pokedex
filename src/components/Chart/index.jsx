import RadarChart from 'react-svg-radar-chart';
import 'react-svg-radar-chart/build/css/index.css';

export default ({ data, size }) => {
    let captions = {};
    let stat = [{
        data: {},
        meta: { color: 'red' }
    }];
    data.map((item, index) => {
        const aux = item.stat.name;
        captions[index] = aux;
        stat[0].data[index] = parseFloat(`0.${item.base_stat}`);
    })
    return <RadarChart data={stat} captions={captions} size={size} />
}