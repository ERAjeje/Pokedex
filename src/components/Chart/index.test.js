import { render } from "@testing-library/react";
import React from "react";
import Chart from ".";

describe('Componente de transação do extrato', () => {
    it('O snapshot do componente deve permanecer sempre o mesmo', () => {
        const data = [
    {
        "base_stat": 45,
        "effort": 0,
        "stat": {
            "name": "hp",
            "url": "https://pokeapi.co/api/v2/stat/1/"
        }
    },
    {
        "base_stat": 49,
        "effort": 0,
        "stat": {
            "name": "attack",
            "url": "https://pokeapi.co/api/v2/stat/2/"
        }
    }
]
        const { container } = render(<Chart data={data} size={450} />);

        expect(container.firstChild).toMatchSnapshot();
    })
})