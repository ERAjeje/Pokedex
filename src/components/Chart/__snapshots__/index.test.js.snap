// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação do extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<svg
  height="405"
  version="1"
  viewBox="-10 0 425 405"
  width="405"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    transform="translate(202.5000,202.5000)"
  >
    <g>
      <circle
        class="scale"
        cx="0"
        cy="0"
        fill="none"
        r="168.75"
      />
      <circle
        class="scale"
        cx="0"
        cy="0"
        fill="none"
        r="112.5"
      />
      <circle
        class="scale"
        cx="0"
        cy="0"
        fill="none"
        r="56.25"
      />
    </g>
    <g>
      <polyline
        class="axis"
        points="0.0000,0.0000 0.0000,-168.7500"
      />
      <polyline
        class="axis"
        points="0.0000,0.0000 0.0000,168.7500"
      />
    </g>
    <g>
      <path
        class="shape "
        d="M0.0000,-75.9375L0.0000,82.6875z"
        fill="red"
        stroke="red"
      />
    </g>
    <g>
      <text
        class="caption"
        dy="5"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        x="0.0000"
        y="-192.3750"
      >
        hp
      </text>
      <text
        class="caption"
        dy="5"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        x="0.0000"
        y="192.3750"
      >
        attack
      </text>
    </g>
  </g>
</svg>
`;
